core/block_svg.js

```diff
@@ -532,6 +532,22 @@ Blockly.BlockSvg.prototype.getBoundingRectangle = function() {
   return new Blockly.utils.Rect(top, bottom, left, right);
 };
 
+
+/**
+ * Toggles the visibility of the block.
+ * @param {boolean} isVisible True if block should be visible.
+ */
+Blockly.BlockSvg.prototype.setVisible = function(isVisible) {
+  var display = isVisible ? 'block' : 'none';
+
+  this.getSvgRoot().style.display = display;
+
+  // Show/hide the inputs.
+  for (var i = 0, input; input = this.inputList[i]; i++) {
+    input.setVisible(isVisible);
```

Comment:
Does this work with collapsed blocks? Collapsing a block also calls setVisible(false) on all of its inputs, if you make them all visible again weird things might happen.

Reply:
Good point, I'll check how the two interface together.